k9mail/src/main/java/com/fsck/k9/Account.java

```diff
@@ -820,6 +820,45 @@ public AccountStats getStats(Context context) throws MessagingException {
         return stats;
     }
 
+    public int getFolderUnreadCount(Context context, String folderName) throws MessagingException {
+        if (!isAvailable(context)) {
+            return 0;
+        }
+
+        int unreadMessageCount = 0;
+        ContentResolver cr = context.getContentResolver();
+
+        Uri uri = Uri.withAppendedPath(EmailProvider.CONTENT_URI,
+                "account/" + getUuid() + "/stats");
+
+        String[] projection = {
+                StatsColumns.UNREAD_COUNT,
+        };
+
+        LocalSearch search = new LocalSearch();
+        search.addAllowedFolder(folderName);
+
+        // Use the LocalSearch instance to create a WHERE clause to query the content provider
+        StringBuilder query = new StringBuilder();
+        List<String> queryArgs = new ArrayList<>();
+        ConditionsTreeNode conditions = search.getConditions();
+        SqlQueryBuilder.buildWhereClause(this, conditions, query, queryArgs);
+
+        String selection = query.toString();
+        String[] selectionArgs = queryArgs.toArray(new String[0]);
```

Comment:
`new String[queryArgs.size()]` prevents Java having to create a pointless array.